### PR TITLE
CONCF-977 | Display correctly cropped thumbnails on Curated Main Page

### PR DESCRIPTION
**Pull request by Warkot**
/cc @rogatty @bognix @vforge

### CODE REVIEW

File: front/scripts/main/components/FeaturedContentItemComponent.ts [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 	style: null,
 
-	// TODO make it more responsive
 	imageWidth: 400,
-	imageHeight: 225,
+	cropMode: Mercury.Modules.Thumbnailer.mode.zoomCrop,
+	thumbUrl: Em.computed('model', function (): string {
+		return this.generateThumbUrl(
+			this.get('model.image_url'),
+			this.get('model.image_crop.' + this.get('aspectRatioName'))
```
Comment (hakubo) [inline front/scripts/main/components/FeaturedContentItemComponent.ts]:
```es6
`model.image_crop.${this.get('aspectRatioName')}`
```

**Comment by hakubo** [comment]
LGTM
apart from tests :)

File: front/scripts/main/mixins/CuratedContentThumbnailMixin.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+		if (imageCropData) {
+			options.mode = this.thumbnailer.mode.windowCrop;
+			options.xOffset1 = imageCropData.x;
+			options.yOffset1 = imageCropData.y;
+			options.xOffset2 = imageCropData.x + imageCropData.width;
+			options.yOffset2 = imageCropData.y + imageCropData.height;
+		} else {
+			options.mode = this.get('cropMode');
+			options.height = this.get('imageHeight') || this.get('imageSize');
```
Comment (rogatty) [inline front/scripts/main/mixins/CuratedContentThumbnailMixin.ts]: What's the `imageSize` for? I think we should decide on a single way to handle this.

@Warkot you've added `block` parameter to all components, so if they have `imageWidth` we calculate `imageHeight` based on that. I'd remove all instances of `imageSize`.

**Comment by rogatty** [comment]
`PhantomJS 1.9.8 (Linux 0.0.0): Executed 78 of 135 (4 FAILED)`

File: front/scripts/main/components/CuratedContentItemComponent.ts [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
+/// <reference path="../mixins/ViewportMixin.ts" />
+/// <reference path="../../mercury/modules/Thumbnailer.ts" />
 'use strict';
 
-App.CuratedContentItemComponent = Em.Component.extend(App.ViewportMixin, {
+App.CuratedContentItemComponent = Em.Component.extend(
+	App.CuratedContentThumbnailMixin,
+	App.ViewportMixin,
+{
+	block: 'curated',
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentItemComponent.ts]: Let's move it below Ember's internals (`classNameBindings` is the last one) and group it together with thumbnail related properties.

**Comment by bognix** [comment]
LGTM

File: front/scripts/main/components/CuratedContentItemComponent.ts [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
 	style: null,
-
-	model: null,
-	type: Em.computed.oneWay('model.type'),
 	href: Em.computed.oneWay('model.url'),
+	type: Em.computed.oneWay('model.type'),
+
 	isArticle: Em.computed.equal('model.type', 'article'),
 
+	block: 'curated',
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentItemComponent.ts]: No need to add it as we do `if (block === 'featured')` in `CuratedContentThumbnailMixin`.

Comment (Warkot) [inline front/scripts/main/components/CuratedContentItemComponent.ts]: I wanted to be explicit. Shouldn't I?

**Comment by tomasznapieralski** [comment]
LGTM

File: front/scripts/main/components/FeaturedContentItemComponent.ts [hunk 5]
```diff
@@ -1,22 +1,28 @@
...
-	cropMode: Mercury.Modules.Thumbnailer.mode.zoomCrop,
-	emptyGif: 'data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAQAIBRAA7',
-	imageUrl: Em.computed.oneWay('emptyGif'),
-	href: Em.computed.oneWay('url'),
+	classNames: ['featured-content-item'],
 	style: null,
+	href: Em.computed.oneWay('model.article_local_url'),
 
-	// TODO make it more responsive
+	block: 'featured',
```
Comment (rogatty) [inline front/scripts/main/components/FeaturedContentItemComponent.ts]: `aspectRatio`

File: front/scripts/main/components/FeaturedContentItemComponent.ts [hunk 6]
```diff
@@ -1,22 +1,28 @@
...
 	imageWidth: 400,
-	imageHeight: 225,
+	cropMode: Mercury.Modules.Thumbnailer.mode.zoomCrop,
```
Comment (rogatty) [inline front/scripts/main/components/FeaturedContentItemComponent.ts]: This is irrelevant when we're sending crop data to `CuratedContentThumbnailMixin#generateThumbUrl`

Comment (Warkot) [inline front/scripts/main/components/FeaturedContentItemComponent.ts]: It is relevant if you want to create a thumbnail with `image_crop` data, but it's not there. Then you want to create a default thumbnail but not in `topCrop` mode but in a `zoomCrop` mode.

Comment (rogatty) [inline front/scripts/main/components/FeaturedContentItemComponent.ts]: Yup, you're right!

File: test/specs/front/scripts/main/components/InfoboxHeroImageMediaComponent.js [hunk 7]
```diff
@@ -1,4 +1,16 @@
...
+	setup: function () {
+		Mercury.Modules.Thumbnailer.getThumbURL = function (url, options) {
+			return url + '/' + options.mode + '/' + options.width + '/' + options.height;
+		}
+	},
+
+	teardown: function () {
+		Mercury.Modules.Thumbnailer.getThumbURL = originalThumbnailerGetThumbURL;
+	}
+});
```
Comment (rogatty) [inline test/specs/front/scripts/main/components/InfoboxHeroImageMediaComponent.js]: /cc @dianafa 

Something to have in mind in the future. You were relating on another test to mock this function and after removing https://github.com/Wikia/mercury/pull/1196/files#diff-e1b839e7d33d61bc6cfb82b68d9c7517L40 this one started to fail.

**Comment by rogatty** [comment]
LGTM

File: test/specs/front/scripts/main/components/InfoboxHeroImageMediaComponent.js (hunk 7 above)
Comment (dianafa) [inline test/specs/front/scripts/main/components/InfoboxHeroImageMediaComponent.js]: thank you! :+1:

**Comment by vforge** [comment]
LGTM

**Pull request merged**